cli/concept.md

```diff
@@ -0,0 +1,157 @@
+# Overview
+The Amplify cli enables front-end developers to easily set up the backend resources in the cloud. <br/>
+It is designed to work side by side with the [Amplify](https://github.com/aws-amplify/amplify-js) library. Official resource categories provided by the cli can be easily consumed by the corresponding Amplify library modules. <br/>
+The Amplify cli is written in Node.js. It has a pluggable architecture and can be easily extended with additional functionalities.
+ 
+***
+
+# Architecture
+![](https://github.com/UnleashedMind/docs/blob/master/cli/images/AmplifyCliConcept.jpg)
+## Cli core and plugins
+The Amplify cli uses `gluegun`. It is highly modularized.  <br/>
+The cli core provides the pluggable platform, most of the cli functionalities are delivered by plugins. <br/> 
+The cli core searches for plugins in the global `node_modules` directory, and its own `node_modules` directory.  <br/>
+Plugins are recognized by the `amplify-` prefix in the package names. <br/>
+Plugins communicate with the cli core, and with each other, through the project metadata. The cli core provides the read and write access to the project metadata for the plugins. The project metadata is stored in file `amplify/backend/amplify-meta.json` in the user project.
```

Comment:
Do we have a link to the project meta data? If so please link otherwise briefly describe it here.